src/ctim/schemas/vulnerability.cljc

```diff
@@ -265,6 +271,71 @@
   (f/required-entries
    (f/entry :cve_data_meta CVEDataMeta)))
 
+;; Link to the spec - https://csrc.nist.gov/schema/cpe/2.3/cpe-dictionary_2.3.xsd
+(def cpe-23-string
+  #"cpe:2\.3:[aho\*\-](:(((\?*|\*?)([a-zA-Z0-9\-\._]|(\\[\\\*\?!\"#$$%&'\(\)\+,:\;<=>@\[\]\^`\{\|}~]))+(\?*|\*?))|[\*\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\*\-]))(:(((\?*|\*?)([a-zA-Z0-9\-\._]|(\\[\\\*\?!\"#$$%&'\(\)\+,:;<=>@\[\]\^`\{\|}~]))+(\?*|\*?))|[\*\-])){4}")
+
+(def cpe-23-string-pattern
+  (re-pattern cpe-23-string))
+
+(def cpe-node-operators
+  #{"OR" "AND"})
+
+(def-enum-type cpe-node-operator-string
+  cpe-node-operators
+  :description (str "The operator string influences how seqs of cpe matches"
+                    "are related to one another.")
+  :gen (spec/gen cpe-node-operators))
+
+(defn cpe-23-string?
+  "Validate a CPE 2.3 string using a regexp"
+  [s]
+  (some? (re-matches cpe-23-string-pattern s)))
+
+(def CPE23String
+  (f/str
+   :description "A text representation of a software or hardware platform."
+   :spec cpe-23-string?
+   #?(:clj :gen)
+   #?(:clj gen/cpe-23-string)))
+
+(def Version
+  (f/str
+   :description (str "A string representing a  bound of"
+                     "a version in the CPE.")
+   :spec (spec/and string? (pred/max-len 64))
+   #?(:clj :gen)
+   #?(:clj gen/semver)))
+
+(def-map-type CPEMatch
+  (concat
+   (f/required-entries
+    (f/entry :vulnerable f/any-bool)
+    (f/entry :cpe23Uri CPE23String))
+   (f/optional-entries
+    (f/entry :versionStartIncluding Version)
```

Comment:
```suggestion
    (f/entry :versionStartIncluding Version
             :description (str "A string representing the lower bound of version included"
                     "in the CPE."))
```

It's a best practice to add a description to all fields. It is uses by `Flanders` generators (ex: GraphQL). Here is an [example](https://github.com/threatgrid/ctim/blob/master/src/ctim/schemas/malware.cljc#L24). You can easily use descriptions from the XML schema https://csrc.nist.gov/schema/cpe/2.3/cpe-dictionary_2.3.xsd